File: src/main/java/com/example/jwttest/dto/ErrorDto.java
package com.example.jwttest.dto;

public class ErrorDto {

    private final int status;
    private final String message;

    public ErrorDto(int status, String message) {
        this.status = status;
        this.message = message;
    }

    public int getStatus() {
        return this.status;
    }

    public String getMessage() {
        return this.message;
    }
}

File: src/main/kotlin/com/example/jwttest/dto/ErrorDto.kt
package com.example.jwttest.dto

import org.springframework.validation.FieldError

class ErrorDto(val status: Int, val message: String?) {
    private val fieldErrors: MutableList<FieldError> = ArrayList()
    fun addFieldError(objectName: String?, path: String?, message: String?) {
        val error = FieldError(objectName!!, path!!, message!!)
        fieldErrors.add(error)
    }

    val fieldError: List<FieldError>
        get() = fieldErrors
}